EIPS/eip-6239.md

```diff
@@ -0,0 +1,170 @@
+
+updateRDF (uint256 tokenId, RDFData rdfData): This **OPTIONAL** method is used when it needs to update the RDF data for Semantic SBT. Use this method to find the RDF data for Semantic SBT by tokenId and perform the update. The input RDF data **MUST** conform to W3C RDF standards. When implementing this method, **SHALL** assign updaters as needed, for example, the token issuer can be assigned as an updater. When calling this method, the UpdateRDF event **MUST** be triggered to notify the listener for performing relevant business update.
+
+schemaURI(): This **OPTIONAL** method is used to query the URIs of the schema for the RDF data. RDF Schema is an extension of the basic RDF vocabulary and provides a data-modelling vocabulary for RDF data. It is **RECOMMENDED** to store the RDF Schema in decentralized storage such as Arweave or IPFS. The URIs are then stored in the contract and can be queried by this method.
+
+### Event Specification
+
+CreateRDF: When minting a Semantic Soulbound Token, this event **MUST** be triggered to notify the listener to perform operations with the created RDF data. When calling the event, the input RDF data **MUST** be RDF statements, which are units of information consisting of three parts: a subject, a predicate, and an object.
+
+UpdateRDF: When updating RDF data for a Semantic Soulbound Token, this event **MUST** be triggered to notify the listener to perform update operations accordingly with the updated RDF data. When calling the event, the input RDF data **MUST** be RDF statements, which are units of information consisting of three parts: a subject, a predicate, and an object.
+
+RemoveRDF: When burning or revoking a Semantic Soulbound Token, this event **MUST** be triggered to notify the listener to perform operations with the removed RDF data for the Semantic SBT. When calling the event, the input RDF data **MUST** be RDF statements, which are units of information consisting of three parts: a subject, a predicate, and an object.
+
+## Rationale
+
+The Decentralized Social today is sufficiently under-specified, leaving many methods to be built to reach convergence and interoperability. Continued work for consensus on standard DeSoc methods is needed. In order to manage the huge dataset that will be generated in DeSoc, we need to model the data in an efficient way for better management, easier integration and faster analysis. We introduce the Semantic SBTs, which specify in a common graph data model,as the blockchain-based identity certifications and primitive building blocks in building the structure of Decentralized social to empower the building of social relations and reputations in DeSoc. Thus provides a way to build next-generation social networks.
```

Comment:
This reads more like Motivation-type content. The Rationale section should describe individual technical choices made within the proposal, while the Motivation should justify the existence of the EIP as a whole.

Reply:
We have added a description of the `rdfStatement`.